WARNING:tensorflow:`period` argument is deprecated. Please use `save_freq` to specify the frequency in number of batches seen.
WARNING  `period` argument is deprecated. Please use `save_freq` to specify the frequency in number of batches seen.
2023-04-03 15:09:05.228561: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 4
2023-04-03 15:09:05.228827: I tensorflow/core/grappler/clusters/single_machine.cc:357] Starting new session
2023-04-03 15:09:05.248139: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 78600 MB memory:  -> device: 0, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:17:00.0, compute capability: 8.0
2023-04-03 15:09:05.249126: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 78600 MB memory:  -> device: 1, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:65:00.0, compute capability: 8.0
2023-04-03 15:09:05.250177: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:2 with 78600 MB memory:  -> device: 2, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:ca:00.0, compute capability: 8.0
2023-04-03 15:09:05.251112: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:3 with 78600 MB memory:  -> device: 3, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:e3:00.0, compute capability: 8.0
2023-04-03 15:09:05.278949: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1137] Optimization results for grappler item: graph_to_optimize
  function_optimizer: function_optimizer did nothing. time = 0.032ms.
  function_optimizer: function_optimizer did nothing. time = 0.002ms.
WARNING:tensorflow:From /home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/tensorflow/python/ops/math_ops.py:3777: tensor_shape_from_node_def_name (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.tensor_shape_from_node_def_name`
WARNING  From /home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/tensorflow/python/ops/math_ops.py:3777: tensor_shape_from_node_def_name (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.tensor_shape_from_node_def_name`
2023-04-03 15:09:05.532762: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
Epoch 1/10
2023-04-03 15:09:09.319299: I tensorflow/stream_executor/cuda/cuda_blas.cc:1760] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
37/37 - 5s - loss: 0.5076 - accuracy: 0.8055 - auc: 0.5158 - precision: 0.2319 - recall: 0.0378 - balanced_accuracy: 0.5057 - val_loss: 0.4516 - val_accuracy: 0.8209 - val_auc: 0.6335 - val_precision: 0.0000e+00 - val_recall: 0.0000e+00 - val_balanced_accuracy: 0.5000
Epoch 2/10
37/37 - 0s - loss: 0.4761 - accuracy: 0.8165 - auc: 0.5838 - precision: 0.3778 - recall: 0.0401 - balanced_accuracy: 0.5138 - val_loss: 0.4364 - val_accuracy: 0.8252 - val_auc: 0.6723 - val_precision: 1.0000 - val_recall: 0.0236 - val_balanced_accuracy: 0.5110
Epoch 3/10
37/37 - 0s - loss: 0.4545 - accuracy: 0.8249 - auc: 0.6339 - precision: 0.5865 - recall: 0.0720 - balanced_accuracy: 0.5304 - val_loss: 0.4251 - val_accuracy: 0.8370 - val_auc: 0.7005 - val_precision: 1.0000 - val_recall: 0.0896 - val_balanced_accuracy: 0.5502
Epoch 4/10
37/37 - 0s - loss: 0.4395 - accuracy: 0.8283 - auc: 0.6772 - precision: 0.6133 - recall: 0.1086 - balanced_accuracy: 0.5475 - val_loss: 0.4156 - val_accuracy: 0.8395 - val_auc: 0.7194 - val_precision: 0.9231 - val_recall: 0.1132 - val_balanced_accuracy: 0.5596
Epoch 5/10
37/37 - 0s - loss: 0.4301 - accuracy: 0.8372 - auc: 0.6854 - precision: 0.7159 - recall: 0.1488 - balanced_accuracy: 0.5674 - val_loss: 0.4101 - val_accuracy: 0.8404 - val_auc: 0.7281 - val_precision: 0.9259 - val_recall: 0.1179 - val_balanced_accuracy: 0.5615
Epoch 6/10
37/37 - 0s - loss: 0.4179 - accuracy: 0.8361 - auc: 0.7165 - precision: 0.6749 - recall: 0.1617 - balanced_accuracy: 0.5722 - val_loss: 0.4039 - val_accuracy: 0.8454 - val_auc: 0.7395 - val_precision: 0.9394 - val_recall: 0.1462 - val_balanced_accuracy: 0.5749
Epoch 7/10
37/37 - 0s - loss: 0.4119 - accuracy: 0.8444 - auc: 0.7224 - precision: 0.7723 - recall: 0.1842 - balanced_accuracy: 0.5856 - val_loss: 0.3993 - val_accuracy: 0.8454 - val_auc: 0.7483 - val_precision: 0.9143 - val_recall: 0.1509 - val_balanced_accuracy: 0.5762
Epoch 8/10
37/37 - 0s - loss: 0.4054 - accuracy: 0.8482 - auc: 0.7338 - precision: 0.7783 - recall: 0.2113 - balanced_accuracy: 0.5970 - val_loss: 0.3952 - val_accuracy: 0.8454 - val_auc: 0.7568 - val_precision: 0.8718 - val_recall: 0.1604 - val_balanced_accuracy: 0.5788
Epoch 9/10
37/37 - 0s - loss: 0.3965 - accuracy: 0.8498 - auc: 0.7471 - precision: 0.7656 - recall: 0.2314 - balanced_accuracy: 0.6068 - val_loss: 0.3919 - val_accuracy: 0.8471 - val_auc: 0.7620 - val_precision: 0.8780 - val_recall: 0.1698 - val_balanced_accuracy: 0.5833
Epoch 10/10
37/37 - 0s - loss: 0.3947 - accuracy: 0.8454 - auc: 0.7532 - precision: 0.7255 - recall: 0.2184 - balanced_accuracy: 0.6010 - val_loss: 0.3888 - val_accuracy: 0.8480 - val_auc: 0.7671 - val_precision: 0.8333 - val_recall: 0.1887 - val_balanced_accuracy: 0.5917
Epoch 00010: val_loss improved from inf to 0.38879, saving model to /home/zadubrov/output/results_train/AR_single-labeled_Fold-0.model.weights.hdf5
<IPython.core.display.HTML object>
INFO     Computation time for training the single-label model for AR: 0.17 min













































































Permutation explainer: 1889it [02:44, 10.90it/s]
Traceback (most recent call last):
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/__main__.py", line 415, in <module>
    main()
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/__main__.py", line 392, in main
    train(fixed_opts)
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/__main__.py", line 190, in train
    sl.train_single_label_models(df=df, opts=opts)
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/single_label_model.py", line 551, in train_single_label_models
    performance = fit_and_evaluate_model(x_train=x_train, x_test=x_test, y_train=y_train, y_test=y_test,
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/single_label_model.py", line 422, in fit_and_evaluate_model
    sd.shap_explain(x_train, model, target, opts.outputDir, drop_values=True, threshold=10, save_values=True)
  File "/gpfs0/home/zadubrov/deepFPlearn_exai/dfpl/shap_dfpl.py", line 28, in shap_explain
    np.savetxt(idx, x_include_labels, fmt="%d", delimiter=";")
  File "<__array_function__ internals>", line 5, in savetxt
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/numpy/lib/npyio.py", line 1367, in savetxt
    X = np.asarray(X)
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/numpy/core/_asarray.py", line 83, in asarray
    return array(a, dtype, copy=False, order=order)
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/shap/_explanation.py", line 421, in __getitem__
    new_self._s = new_self._s.__getitem__(item)
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/slicer/slicer.py", line 112, in __getitem__
    sliced_o = tracked[slicer_index]
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/slicer/slicer_internal.py", line 69, in __getitem__
    return UnifiedDataHandler.slice(self.o, index_tup, self.max_dim)
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/slicer/slicer_internal.py", line 583, in slice
    out = tail_slice(sliced_o, index_tup[cut:], max_dim - cut, is_element)
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/slicer/slicer_internal.py", line 544, in tail_slice
    results = [AtomicSlicer(e, max_dim=max_dim)[tail_index] for e in o]
  File "/home/zadubrov/.conda/envs/dfpl_exai/lib/python3.9/site-packages/slicer/slicer_internal.py", line 544, in <listcomp>
    results = [AtomicSlicer(e, max_dim=max_dim)[tail_index] for e in o]
KeyboardInterrupt